### PR TITLE
Change default renderType based on React version. Resolves #202

### LINKED ISSUE #202: Default renderType should vary based on react version
The default `renderType` is `createRoot`, but that doesn't seem to work with the https://react.microfrontends.app example that's using React 17. To fix, possibly update the `getRenderType` function to do some react version checks?

https://github.com/single-spa/single-spa-react/blob/57f84557c5bf2d49260ec1770e19bd482c83547a/src/single-spa-react.js#L39

https://github.com/single-spa/single-spa-react/blob/57f84557c5bf2d49260ec1770e19bd482c83547a/src/single-spa-react.js#L219

### PR DESCRIPTION
**Pull request by joeldenning**
See #202

### CODE REVIEW

File: src/single-spa-react.js [hunk 1]
```diff
@@ -68,6 +67,15 @@ function singleSpaReact(userOpts) {
     );
   }
 
+  if (!opts.renderType) {
+    // https://github.com/single-spa/single-spa-react/issues/202
+    if (opts.ReactDOM?.createRoot || opts.ReactDOMClient?.createRoot) {
```
Comment (MilanKovacic) [inline src/single-spa-react.js:72]: createRoot is not available in react-dom < 18, so there is no need for that check.

Comment (joeldenning) [reply src/single-spa-react.js]: Good catch.

Didn't some versions before React 18 have `unstable_createRoot`? Regardless, the default should not be something unstable.

**Review by MilanKovacic — APPROVED** [review]
LGTM

File: src/single-spa-react.js (hunk 1 above)
Comment (MilanKovacic) [reply src/single-spa-react.js]: If I remember correctly, that was during the alpha/beta versions of react 18.

**Pull request merged**